src/taipy/core/config/data_node_config.py

```diff
@@ -310,222 +351,232 @@ def _configure(
                 _storage_type_ value set in the default data node configuration
                 (see `(Config.)configure_default_data_node()^`)), _"pickle"_, _"csv"_, _"excel"_, _"sql_table"_,
                 _"sql"_, _"json"_, _"parquet"_, _"mongo_collection"_, _"in_memory"_, or _"generic"_.
-            scope (Scope^): The scope of the data node configuration. The default value is
+            scope (Optional[Scope^]): The scope of the data node configuration. The default value is
                 `Scope.SCENARIO` (or the one specified in
                 `(Config.)configure_default_data_node()^`).
             **properties (Dict[str, Any]): A keyworded variable length list of additional
                 arguments.
         Returns:
             `DataNodeConfig^`: The new data node configuration.
         """
-        section = DataNodeConfig(id, storage_type, scope, **properties)
-        Config._register(section)
-        return Config.sections[DataNodeConfig.name][id]
+        configuration_map: Dict[str, Callable] = {
```

Comment:
This dictionary map the `"storage_type"` string to the corresponding configure method.

Reply:
I wouldn't recommend putting a dictionary as a default value here, this link explains the potential problem in great detail https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python. Of course this only applies if we made modification to the `configuration_map` and we're not doing so in the function so it's probably A-okay, I'll leave the judgement call to you 😃

Reply:
It is dangerous if we set the default value of a function parameter right? `configuration_map` here is a local variable though.

Or do you mean some default values in `_OPTIONAL_PROPERTIES`?

Reply:
ohhh I didn't notice that, you're right!! Nevermind it then :D